eos-phone-home

```diff
@@ -406,66 +400,73 @@ class PhoneHome(object):
     def _do_ping(self):
         if self._send_to_server(self.PING_ENDPOINT, self.PING_VARIABLES):
             if self._debug:
-                print("Debugging turned on so not incrementing count.")
+                log.info("Debugging turned on so not incrementing count.")
             else:
                 # increment the counter upon successful ping only
                 count = self._lookup_or_get_variable('count')
                 self._set_count(count + 1)
 
-            print("Pinged!")
+            log.info("Pinged!")
         else:
-            print("Could not send ping to server!", file=sys.stderr)
+            log.warning("Could not send ping to server!")
 
             return False
 
         return True
 
-    def run(self):
+    def run(self, exit_on_server_error):
         if not os.path.isdir(self.STATE_DIRECTORY):
-            print("State folder (%s) not found! Exiting!" %
-                  self.STATE_DIRECTORY, file=sys.stderr)
+            log.fatal("State folder (%s) not found! Exiting!",
+                      self.STATE_DIRECTORY)
             exit(1)
 
         if not os.access(self.STATE_DIRECTORY, os.W_OK):
-            print("State folder (%s) not writable! Exiting!" %
-                  self.STATE_DIRECTORY, file=sys.stderr)
+            log.fatal("State folder (%s) not writable! Exiting!",
+                      self.STATE_DIRECTORY)
             exit(1)
 
         success = True
 
         try:
             if self._force or self._need_to_activate():
-                print("Activation needed.")
+                log.info("Activation needed.")
                 if not self._do_activate():
                     success = False
             else:
-                print("Already activated!")
+                log.info("Already activated!")
 
             if self._force or self._need_to_ping():
-                print("Ping needed.")
+                log.info("Ping needed.")
                 if not self._do_ping():
                     success = False
             else:
-                print("Ping not due yet.")
-        except Exception as e:
-            print("Unhandled exception:", e, file=sys.stderr)
-            success = False
+                log.info("Ping not due yet.")
+        except:
+            log.exception("Unhandled exception:")
+            exit(1)
 
-        if success:
-            exit(0)
-        else:
+        if not success and exit_on_server_error:
```

Comment:
Will it? In most places, filesystem access was already wrapped in a blanket `try: except:` with a sensible fallback. Writing the `activated` stamp and the ping `count` file will bubble up to the catch-all block which still calls `exit(1)`. I don't think I have suppressed any more filesystem errors that were not previously suppressed.